### PR TITLE
Jw email part 4 request amendment

### PR DESCRIPTION
**Pull request by jwiel86**
For more information, please review the pivotal story:  https://www.pivotaltracker.com/story/show/130746087

### CODE REVIEW

File: app/controllers/service_requests_controller.rb [hunk 1]
```diff
@@ -438,7 +444,10 @@ def remove_service
 
     if ssr.line_items.empty?
       if !ssr.submitted_at.nil?
-        send_ssr_service_provider_notifications(@service_request, ssr, true)
+        ssr_destroyed = true
+        request_amendment = false
+        # only notify service providers of destroyed ssr
+        send_ssr_service_provider_notifications(ssr, ssr_destroyed, request_amendment)
```
Comment (jayhardee9) [inline app/controllers/service_requests_controller.rb]: Couldn't you do: `send_ssr_service_provider_notifications(ssr, true, false)`?

File: app/controllers/service_requests_controller.rb [hunk 2]
```diff
@@ -494,11 +503,32 @@ def new_document
...
+  def send_request_amendment(sub_service_requests)
+    request_amendment = true
+    sub_service_requests = [sub_service_requests].flatten
+    send_service_provider_notifications(sub_service_requests, request_amendment)
+    send_admin_notifications(sub_service_requests, request_amendment)
+  end
+
   # Send notifications to all users.
   def send_notifications(service_request, sub_service_requests)
+    request_amendment = false
```
Comment (jayhardee9) [inline app/controllers/service_requests_controller.rb]: Same comment as above.

File: app/controllers/service_requests_controller.rb [hunk 3]
```diff
@@ -522,35 +552,38 @@ def send_user_notifications(service_request)
     end
   end
 
-  def send_service_provider_notifications(service_request, sub_service_requests) #all sub-service requests on service request
+  def send_service_provider_notifications(sub_service_requests, request_amendment=false) #all sub-service requests on service request
     sub_service_requests.each do |sub_service_request|
-      send_ssr_service_provider_notifications(service_request, sub_service_request)
+      ssr_destroyed = false
```
Comment (jayhardee9) [inline app/controllers/service_requests_controller.rb]: Same here.

File: app/helpers/notifier_helper.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  def display_audit_table(status, audit_report)
+    if status == 'request_amendment' && audit_report.present? && audit_report[:line_items].present?
+      render "audit_action"
+    end
+  end
+
+  def display_notes?(status, role, notes)
+    if (status == "submitted" || status == "request_amendment") && role == 'none' && !notes.empty?
```
Comment (jayhardee9) [inline app/helpers/notifier_helper.rb]: Whole method could just be: `(status == "submitted" || status == "request_amendment") && role == 'none' && !notes.empty?`. No `if` or `return` needed.

Comment (jwiel86) [inline app/helpers/notifier_helper.rb]: Sweet!  thanks!

**Comment by sparc-request** [comment]
I believe could also pass the variable names as you set them.
send_ssr_service_provider_notifications(ssr, ssr_destroyed: true, request_amendment: false)

Assuming that is how they are defined in the method.

Andrew

**Comment by wtholt** [comment]
Rebase this branch to the updated master and re-push

**Pull request merged**